### PR TITLE
[ET-1369] Checkout Front-end: multiple gateways toggle

### PR DESCRIPTION
**Pull request by codingmusician**
### 🎫 Ticket

[ET-1369] 

### 🗒️ Description

In the case where there is more than one gateway enabled and active, we need to show an "Additional payment options" toggle to give customers different options for paying.

### 🎥 Artifacts 

https://i.imgur.com/L0Up1t8.png
https://i.imgur.com/j4x1s9T.png

### ✔️ Checklist
- [ ] I've included a changelog entry in the readme.txt file. 
- [x] My code is tested. 
- [x] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

[ET-1369]: https://theeventscalendar.atlassian.net/browse/ET-1369?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Tickets/Commerce/Gateways/PayPal/Hooks.php [hunk 1]
```diff
@@ -126,46 +123,6 @@ public function include_success_page_vars( $template_vars, $shortcode ) {
...
-	 *
-	 * @since 5.2.0
-	 *
-	 * @param string           $file     Which file we are loading.
-	 * @param string           $name     Name of file file
-	 * @param \Tribe__Template $template Which Template object is being used.
-	 */
-	public function include_advanced_payments( $file, $name, $template ) {
-		$this->container->make( Buttons::class )->include_advanced_payments( $file, $name, $template );
-	}
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/PayPal/Hooks.php:167]: Why was this removed? Why are we touching PayPal codebase?

File: src/Tickets/Commerce/Gateways/PayPal/Buttons.php [hunk 2]
```diff
@@ -78,60 +66,11 @@ public function get_checkout_script() {
...
-		$merchant   = tribe( Merchant::class );
-
-		$template->template(
-			'gateway/paypal/advanced-payments',
-			[
-				'supports_custom_payments' => $merchant->get_supports_custom_payments(),
-				'active_custom_payments'   => $merchant->get_active_custom_payments(),
-				'must_login'               => $must_login,
-			]
-		);
```
Comment (bordoni) [inline src/Tickets/Commerce/Gateways/PayPal/Buttons.php:135]: Why was this removed? Why are we touching PayPal codebase?

File: src/resources/js/commerce/gateway/base.js [hunk 3]
```diff
@@ -80,3 +81,82 @@ tribe.tickets.commerce.billing.getDetails = function( long ) {
...
+			`<button class="${obj.toggler.classes.toggleButton}">` + 
+			`${tecTicketsCommerceCheckoutToggleText.additional}` + 
+			`</button></div>`;
+	};
+	obj.toggler.toggleEvents = () => {
+		obj.toggler.toggles.default.find('button').on( 'click', obj.toggler.showDefault );
+		obj.toggler.toggles.additional.find('button').on( 'click', obj.toggler.showAdditional );
+	}
+	obj.toggler.init();
+})( jQuery, tribe.tickets.commerce.gateway );
```
Comment (bordoni) [inline src/resources/js/commerce/gateway/base.js:162]: Docblocks and new file.

**Review by bordoni — CHANGES_REQUESTED** [review]
If there is a reason for the paypal changes that is totally fine, but I wanted to make sure.

File: src/Tickets/Commerce/Gateways/PayPal/Buttons.php (hunk 2 above)
Comment (codingmusician) [reply src/Tickets/Commerce/Gateways/PayPal/Buttons.php:135]: I consolidated the several different template methods into one `get_checkout_template_vars()` method which returns the variables needed for the templates. Rather than using hooks (in the `Hooks.php` file), I decided to add a main gateways template (`src/views/v2/commerce/gateway/gateways.php`) that calls methods from each gateway to pull the template info to show it. I felt it gave us more flexibility with the templates and reduced the amount of calls it required.

File: src/Tickets/Commerce/Gateways/PayPal/Hooks.php (hunk 1 above)
Comment (codingmusician) [reply src/Tickets/Commerce/Gateways/PayPal/Hooks.php:167]: Same comment from above.

File: src/resources/js/commerce/gateway/base.js (hunk 3 above)
Comment (codingmusician) [reply src/resources/js/commerce/gateway/base.js]: Done. :+1:

**Comment by juanfra** [comment]
I think this PR really needed a thorough review.

**Comment by codingmusician** [comment]
@juanfra If you want to send me some notes on changes you'd like to see made, let me know and I'll whip up a new PR for them.

File: src/views/v2/commerce/checkout.php [hunk 4]
```diff
@@ -40,21 +40,8 @@ class="tribe-tickets__commerce-checkout"
...
-				'notice_classes'  => [
-						'tribe-tickets__notice--error',
-						'tribe-tickets__commerce-checkout-notice',
-				],
-				'content_classes' => [ 'tribe-tickets__commerce-checkout-notice-content' ],
-				'title'           => __( 'Checkout Error!', 'event-tickets' ),
-				'content'         => __( 'Something went wrong!', 'event-tickets' ),
-			]
-		);
-		?>
```
Comment (rafsuntaskin) [inline src/views/v2/commerce/checkout.php:57]: @codingmusician @juanfra is this removed intentionally? The notice component is not getting loaded into the checkout page anymore.
